share/doc/wit/tutorial.md

```diff
@@ -268,3 +268,35 @@ wit status
 
 There we have it! We have made package `foo` depend on package `bar`.
 
+
+## Using a local repository cache for a faster 'wit init'
```

Comment:
Nice docs!

I think we may want this to live somewhere besides the tutorial, since I wouldn't consider it "essential" enough for an introduction tutorial to Wit. I had created a ["how-to guides"](https://github.com/sifive/wit/blob/261f9c3af98302aa2f58b3fbed66586047457273/docs/how-to-guides.adoc) document, which is an unordered collection of useful recipes, although I now realize is probably in the wrong directory (`wit/docs/` instead of `wit/share/doc/wit`).

How about we move these docs there, and then possibly move the how-to-guides into the `share/` directory, either in this PR or in a followup PR?

Reply:
I kinda think a top level docs directory is way more discoverable than share/doc

Reply:
moved it all into `Documentation/`

Reply:
`/share/doc/wit` was following the Debian standard. I don't really care about following that, but if we're going to move it the top, I think `doc/` is more common and consistent with the other directory names (eg. `lib`)